tar/tar.go

```diff
@@ -155,6 +155,12 @@ func createAndFill(filePath string, mode int64, content io.Reader) error {
 	if err != nil {
 		return fmt.Errorf("cannot set proper mode on file %q: %v", filePath, err)
 	}
+	if err := fh.Sync(); err != nil {
```

Comment:
I started to say that it would be better to put this and the close into the defer and make them avoid clobbering any other errors being returned while still returning any errors from them, but actually it's unwieldy enough that I don't think it's worth doing.